### DISCUSSION TITLE
[Feature] mobile app: share/upload from immich

### DISCUSSION THREAD

**Discussion by hofq** [immich-app/immich#11555] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Counter-Part to #2659

Personally i am not a fan of the  Android Gallery Structure and find it really hard to Find Pictures in it.

To upload a picture to an app, sometimes the app includes a simple selector for selecting a recent picture to upload. Sometimes The App has not implemented this itself and allows you to choose a gallery app to select your pictures. If that accurs i would love to be able to select immich and select images that are stored on my device/ on the server.

For that a few things should be added:
- Support the "Select"/ "Upload" Function on Devices (I know only for android that it exists)
- Support Offline Mode for Seing the local gallery
- Dynamically download selected pictures from server in the Select Menu

It would be awesome if the explore tab would also be usable in that menu, would be a game changer for finding pictures to be uploaded.

![Screenshot_20240804-015049](https://github.com/user-attachments/assets/6630cdd6-c905-46bf-a80c-0a5a482ddf3b)

![Screenshot_20240804-015053(1)](https://github.com/user-attachments/assets/63a0f129-37e4-4ab3-be4c-0d56495b377b)

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Only one feature request per discussion please. Some of these are also duplicates.
